2023/src/day7/camel_hand_parse.rs

```diff
@@ -0,0 +1,40 @@
+
+use super::playing_card::{CamelHand, PlayingCard, get_hand_type};
+
+pub fn parse_camel_hands(input_lines: &Vec<String>) -> Vec<CamelHand> {
+  let mut camel_hands = Vec::new();
+  for input_line in input_lines.iter() {
+    let hand_parts: Vec<&str> = input_line.split(" ")
+      .map(|hand_part| hand_part.trim())
+      .collect();
+    let cards:Vec<PlayingCard> = hand_parts[0].split("")
+      .filter(|card_str| card_str.len() > 0)
+      .map(|card_str| match card_str {
+        "A" => PlayingCard::Ace,
+        "K" => PlayingCard::King,
+        "Q" => PlayingCard::Queen,
+        "J" => PlayingCard::Jack,
+        "T" => PlayingCard::Ten,
+        "9" => PlayingCard::Nine,
+        "8" => PlayingCard::Eight,
+        "7" => PlayingCard::Seven,
+        "6" => PlayingCard::Six,
+        "5" => PlayingCard::Five,
+        "4" => PlayingCard::Four,
+        "3" => PlayingCard::Three,
+        "2" => PlayingCard::Two,
+        _ => panic!("Invalid playing card: {}", card_str)
```

Comment:
I like throwing `unreachable!()` in places like this.  It still panics, just another readability/messaging thing.

Reply:
good catch, I learned about `unreachable!()` some time after this :) p. nifty